lib/gems/pending/util/require_with_logging.rb

```diff
@@ -24,22 +26,33 @@ module Kernel
   }
   REQ_LOG_TREE = "| "
 
-  def log_require(path, mode, timing = nil)
+  def require_with_metrics
+    initial_memory = Sys::ProcTable.ps(:pid => Process.pid).rss
+    initial_loaded = $LOADED_FEATURES.length
+
+    result = yield
+
+    memory_after = Sys::ProcTable.ps(:pid => Process.pid).rss
+    loaded_after = $LOADED_FEATURES.length
+    return result, (memory_after - initial_memory), (loaded_after - initial_loaded)
+  end
+
+  def log_require(path, mode, timing = nil, memory_consumed = nil, files_loaded = nil)
     $req_depth -= 1 if mode == true || mode == false || mode == :fail
-    $req_log.puts "#{$req_depth.to_s.rjust(3)}  #{REQ_LOG_OPERS[mode]}  #{REQ_LOG_TREE * $req_depth}#{path.inspect[1..-2]}#{timing.nil? ? '' : " (#{"%.6f" % timing})"}"
+    $req_log.puts "#{$req_depth.to_s.rjust(3)}  #{REQ_LOG_OPERS[mode]}  #{REQ_LOG_TREE * $req_depth}#{path.inspect[1..-2]}#{timing.nil? ? '' : " (#{"%.6f" % timing})"}#{memory_consumed.nil? ? '' : " (#{memory_consumed})"}#{files_loaded.nil? ? '' : " (#{files_loaded})"}"
```

Comment:
I think this needs units now to distinguish the numbers

```suggestion
    $req_log.puts "#{$req_depth.to_s.rjust(3)}  #{REQ_LOG_OPERS[mode]}  #{REQ_LOG_TREE * $req_depth}#{path.inspect[1..-2]}#{timing.nil? ? '' : " (#{"%.6f" % timing}s)"}#{memory_consumed.nil? ? '' : " (#{memory_consumed / 1024.0}MB)"}#{files_loaded.nil? ? '' : " (#{files_loaded} files)"}"
```

Reply:
I noticed also that this says "files loaded", but earlier it's loaded features - is that the same thing?  Should it say "features loaded" instead?

Reply:
sure, I can do that

Reply:
ok, done, used KB, see below.